services/web/server/tests/unit/with_dbs/02/test_projects_nodes_handlers__patch.py

```diff
@@ -0,0 +1,152 @@
+# pylint: disable=protected-access
+# pylint: disable=redefined-outer-name
+# pylint: disable=too-many-arguments
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+# pylint: disable=too-many-statements
+
+
+import json
+from http import HTTPStatus
+
+import pytest
+from aiohttp.test_utils import TestClient
+from pytest_mock.plugin import MockerFixture
+from pytest_simcore.helpers.utils_assert import assert_status
+from pytest_simcore.helpers.utils_login import UserInfoDict
+from servicelib.aiohttp import status
+from simcore_service_webserver._meta import api_version_prefix
+from simcore_service_webserver.db.models import UserRole
+from simcore_service_webserver.projects.models import ProjectDict
+
+API_PREFIX = "/" + api_version_prefix
+
+
+@pytest.fixture
+def mock_catalog_api_get_services_for_user_in_product(mocker: MockerFixture):
+    mocker.patch(
+        "simcore_service_webserver.projects._crud_handlers.get_services_for_user_in_product",
+        spec=True,
+        return_value=[],
+    )
+
+
+@pytest.fixture
+def mock_project_uses_available_services(mocker: MockerFixture):
+    mocker.patch(
+        "simcore_service_webserver.projects._crud_handlers.project_uses_available_services",
+        spec=True,
+        return_value=True,
+    )
+
+
+@pytest.mark.parametrize(
+    "user_role,expected",
+    [
+        (UserRole.ANONYMOUS, status.HTTP_401_UNAUTHORIZED),
+        (UserRole.GUEST, status.HTTP_403_FORBIDDEN),
+        (UserRole.USER, status.HTTP_204_NO_CONTENT),
+        (UserRole.TESTER, status.HTTP_204_NO_CONTENT),
+        (UserRole.ADMIN, status.HTTP_204_NO_CONTENT),
+        (UserRole.PRODUCT_OWNER, status.HTTP_204_NO_CONTENT),
+    ],
+)
+async def test_patch_project_node_access_rights(
```

Comment:
same here. this is testing the entrypoint_access.

And the fixture I was thinking of is ```standard_role_response``` in [https://github.com/ITISFoundation/osparc-simcore/blob/1e128a9fe50b651938addafcd74b92fc3f81b2e1/packages/pytest-simcore/src/pytest_simcore/helpers/utils_webserver_unit_with_db.py](https://github.com/ITISFoundation/osparc-simcore/blob/1e128a9fe50b651938addafcd74b92fc3f81b2e1/packages/pytest-simcore/src/pytest_simcore/helpers/utils_webserver_unit_with_db.py)